internal/uploader.go

```diff
@@ -32,6 +34,7 @@ type Uploader interface {
 	Clone() Uploader
```

Comment:
What happens if we Clone the uploader? AFAIK tarSize and dataSize are shared between different uploaders. I guess that this statistics routine should also be the same for different uploaders?